source/postprocess/boundary_velocity_residual_statistics.cc

```diff
@@ -55,6 +62,23 @@ namespace aspect
 
           gplates_lookup->load_file(data_directory + data_file_name, this->get_mpi_communicator());
         }
+
+    }
+
+
+
+    template <int dim>
+    Point<dim>
+    BoundaryVelocityResidualStatistics<dim>::get_spherical_position (const Point<dim> &p) const
+    {
+      const std::array<double,dim> spherical_position = this->get_geometry_model().
+                                                        cartesian_to_natural_coordinates(p);
+      Point<dim> internal_position = p;
+
+      for (unsigned int d = 0; d < dim; ++d)
+        internal_position[d] = spherical_position[d];
+
+      return internal_position;
     }
```

Comment:
Unless you need this function in a few other places as well, I think it does not make things simpler here (it saves about 3-5 lines in line 128, but adds this whole function instead). Also at the moment this function is public, but it is not used outside of this class. I would remove it.